similarity_search/CMakeLists.txt

```diff
@@ -59,7 +59,7 @@ elseif(${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
     endif()
     if (CMAKE_SYSTEM_NAME MATCHES Darwin)
         # MACOSX
-        set (CMAKE_CXX_FLAGS_RELEASE "-Wall -Wunreachable-code -Wcast-align -O3 -DNDEBUG -std=c++11 -DHAVE_CXX0X -fpic")
+        set (CMAKE_CXX_FLAGS_RELEASE "-Wall -Wunreachable-code -Wcast-align -O3 -DNDEBUG -std=c++11 -DHAVE_CXX0X -fpic -msse4.2")
```

Comment:
@benfred I will sure merge this, but I really wonder if -march=native works on MAC OSX. Currently, it's hard to find a machine without AVX2 and some code depends on AVX2 (there will be more code doing so soon).